test/e2e/create_test.go

```diff
@@ -66,21 +66,27 @@ func testWindowsNodeCreation(t *testing.T) {
 		// provisioned and then change the key. The the correct amount of nodes being configured is proof that the
 		// mismatched Machine created with the mismatched key was deleted and replaced.
 		// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-		// the first configured Machines to hit this scenario.
+		// the first configured Machines to hit this scenario. This is a platform agonistic test so we run it only on
+		// AWS.
 		err = testCtx.waitForWindowsMachines(int(gc.numberOfNodes), "Provisioned")
 		require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-		err = testCtx.createPrivateKeySecret(false)
-		require.NoError(t, err, "error replacing private key secret")
+		if testCtx.CloudProvider.GetType() == config.AWSPlatformType {
+			// Replace the known private key with a randomly generated one.
+			err = testCtx.createPrivateKeySecret(false)
+			require.NoError(t, err, "error replacing private key secret")
+		}
 		err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, false, false)
 		assert.NoError(t, err, "Windows node creation failed")
 
-		t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
-			// Replace private key and check that new Machines are created using the new private key
-			err = testCtx.createPrivateKeySecret(true)
-			require.NoError(t, err, "error replacing private key secret")
-			err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, false, false)
-			assert.NoError(t, err, "error waiting for Windows nodes configured with newly created private key")
-		})
+		if testCtx.CloudProvider.GetType() == config.AWSPlatformType {
+			t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
```

Comment:
You can do the inverse of this and use t.Skip() to reduce indentation.
```suggestion
	t.Run("Changing the private key ensures all Windows nodes use the new private key", func(t *testing.T) {
		if testCtx.CloudProvider.GetType() != config.AWSPlatformType {
			t.Skip()
		}
...
```
This also allows us to be verbose about skipping the test on a platform in the test report.